### DISCUSSION TITLE
Disabling compiler warnings?

### DISCUSSION THREAD

**Discussion by TriceHelix** [corrosion-rs/corrosion#502] [Q&A] [answered]
Hi everyone,
I'm currently trying to disable compiler warnings for a third-party crate.
Because Corrosion seems to need at least one compilable target in the crate manifest, I cannot simply add `#![allow(warnings)]` to some kind of "meta" manifest file which pulls in my dependency. I have tried adjusting the configuration by passing --config to the FLAGS argument of the corrosion_import_crate() function, but that doesn't really seem to work either. (Probably because I am using it incorrectly?)
I'm kind of a noob when it comes to Rust in general (hence why I'm using Corrosion in the first place). Any kind of help/advice would be highly appreciated!

**Comment by jschwe** [comment, marked as answer]
>  I'm currently trying to disable compiler warnings for a third-party crate.

I think the best solution here is to open an issue or PR upstream in the repository of the crate, and ask them to fix the warning.

> I cannot simply add #![allow(warnings)] to some kind of "meta" manifest file which pulls in my dependency.

This is independent of corrosion, and doesn't work with pure cargo builds either. `#![allow(warnings)]` are placed at the top of a `.rs` file and affect everything below. External crates are not affected by this.

In theory, you could use `RUSTFLAGS` to globally silence warnings (add `--allow=warnings`, either via a `.cargo/config.toml` file or via [corrosion_add_target_rustflags(https://corrosion-rs.github.io/corrosion/usage.html#per-target-options)] ). But I would strongly recommend against globally silencing warnings and instead try to fix the warning upstream.

**Reply by TriceHelix** [reply]
Thank you, especially [this](https://corrosion-rs.github.io/corrosion/usage.html#per-target-options) helped a lot.
As you can tell I wasn't really aware of how corrosion works under the hood (or how exactly it invokes cargo), but it makes a lot more sense to me now. [This stackoverflow answer](https://stackoverflow.com/a/25877389) also goes into detail on how to accomplish what I'm trying to do in general, and utilizing Corrosion's `corrosion_add_target_rustflags` in my CMake config worked flawlessly.

PS: As far as I can tell, the warnings I'm receiving are harmless, and since the package stems from the Unicode Consortium, I'm confident that suppressing them to clean up the console output will not produce horrible side-effects.
